### PR TITLE
Feature/new arena scoring system

### PR DESCRIPTION
**Pull request by sonohoshi**
### Description

In the past, defender in arena didn't get any score.
This is a new scoring system of defender.

### Related Links
[아레나 스코어 로직 수정 (진사람 점수 까기)](https://app.asana.com/0/958521740385861/1201406778349280/f)
[2021-11-22 9C 주간](https://www.notion.so/planetarium/2021-11-22-9C-7a0c525e580c4e87bbd2d904da6a26bf#acc7d16643c44d84950bfd58d30cdf29)

### CODE REVIEW

File: Lib9c/Action/RankingBattle8.cs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+            if (ctx.Rehearsal)
+            {
+                return states
+                    .SetState(avatarAddress, MarkChanged)
+                    .SetState(weeklyArenaAddress, MarkChanged)
+                    .SetState(inventoryAddress, MarkChanged)
+                    .SetState(worldInformationAddress, MarkChanged)
+                    .SetState(questListAddress, MarkChanged);
+            }
+
```
Comment (hskim881028) [inline Lib9c/Action/RankingBattle8.cs:54]:
```suggestion
CheckObsolete(BlockChain.Policy.BlockPolicySource.V100089ObsoleteIndex, context);
```
you need to add that code

**Review by hskim881028 — CHANGES_REQUESTED** [review]

File: Lib9c/Battle/ArenaScoreHelper.cs [hunk 2]
```diff
@@ -80,28 +81,49 @@ public static class ArenaScoreHelper
...
+                (-100, 25, -8, 1, -1),
+                (0, 20, -8, 1, -1),
+                (100, 15, -10, 1, -1),
+                (200, 15, -10, 0, 0),
+                (300, 8, -20, 0, 0),
+                (400, 4, -25, 0, 0),
+                (500, 2, -30, 0, 0),
+            }.OrderBy(tuple => tuple.differ);
+
+        public static int GetScore(int challengerRating, int defenderRating, BattleLog.Result result, out int defenderScore)
```
Comment (hskim881028) [inline Lib9c/Battle/ArenaScoreHelper.cs:105]: It seems nice to use a tuple to return both me and the enemy score.

**Review by hskim881028 — APPROVED** [review]

**Comment by Namyujeong** [comment]
image

**Comment by boscohyun** [comment]
Fix in #764

**Pull request closed without merging**